pallets/asset/src/identifier.rs

```diff
@@ -0,0 +1,143 @@
+use codec::{Decode, Encode};
+use std::thread::sleep_ms;
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq)]
+pub enum Identifier {
+    CUSIP([u8; 9]),
+    CINS([u8; 9]),
+    ISIN([u8; 12]),
+    LEI([u8; 20]),
+}
+
+impl Identifier {
+    pub fn cusip(bytes: [u8; 9]) -> Option<Identifier> {
+        if cusip_checksum(&bytes[..8]) == bytes[8] - b'0' {
```

Comment:
Deserves a comment I think (other functions below as well).
Also, nit: this is a nice candidate for https://doc.rust-lang.org/nightly/std/primitive.bool.html#method.then_some.

Reply:
Looks like it's nightly only

Reply:
Not a problem though; we can and do use nightly features.

Reply:
FYI substrate already using nightly features multiple please, so I don't see any problem to use it here as well